src/Core/src/Platform/Android/SliderExtensions.cs

```diff
@@ -47,37 +50,62 @@ public static void UpdateThumbColor(this SeekBar seekBar, ISlider slider) =>
 		public static async Task UpdateThumbImageSourceAsync(this SeekBar seekBar, ISlider slider, IImageSourceServiceProvider provider)
 		{
 			var context = seekBar.Context;
-
-			if (context == null)
+			if (context is null || !seekBar.IsAlive())
 				return;
 
 			var thumbImageSource = slider.ThumbImageSource;
-
-			if (thumbImageSource != null)
+			if (thumbImageSource is not null)
 			{
 				var service = provider.GetRequiredImageSourceService(thumbImageSource);
 				var result = await service.GetDrawableAsync(thumbImageSource, context);
-
 				var thumbDrawable = result?.Value;
 
-				if (seekBar.IsAlive() && thumbDrawable != null)
-					seekBar.SetThumb(thumbDrawable);
-			}
-			else
-			{
-				seekBar.SetThumb(context.GetDrawable(Resource.Drawable.abc_seekbar_thumb_material));
-				if (slider.ThumbColor is null && context.Theme is not null)
+				if (thumbDrawable is not null)
 				{
-					using var value = new TypedValue();
-					context.Theme.ResolveAttribute(Android.Resource.Attribute.ColorAccent, value, true);
-					var color = new Color(value.Data);
-					seekBar.Thumb?.SetColorFilter(color, FilterMode.SrcIn);
+					SetThumbDrawable(seekBar, context, thumbDrawable);
 				}
 				else
 				{
-					seekBar.UpdateThumbColor(slider);
+					SetDefaultThumb(seekBar, slider, context);
 				}
 			}
+			else
+			{
+				SetDefaultThumb(seekBar, slider, context);
+			}
+		}
+
+		static void SetThumbDrawable(SeekBar seekBar, Android.Content.Context context, Drawable thumbDrawable)
+		{
+			//Material 2 design spec - https://m2.material.io/components/sliders/android#discrete-slider
+			const int TARGET_SIZE = 20; // 10 radius * 2
+			int thumbSize = (int)context.ToPixels(TARGET_SIZE);
+
+			Bitmap bitmap = Bitmap.CreateBitmap(thumbSize, thumbSize, Bitmap.Config.Argb8888!);
+			Canvas canvas = new Canvas(bitmap);
+			thumbDrawable.SetBounds(0, 0, thumbSize, thumbSize);
+			thumbDrawable.Draw(canvas);
+
+			BitmapDrawable finalDrawable = new BitmapDrawable(context.Resources, bitmap);
```

Comment:
Used using statements to ensure proper disposal of BitmapDrawable:
```
using (BitmapDrawable finalDrawable = new BitmapDrawable(context.Resources, bitmap))
{
    seekBar.SetThumb(finalDrawable);
}
```

Reply:
@jsuarezruiz ,  I have updated the code changes. please let me know if you have any concerns.